src/ck3/data/lease_contracts.rs

```diff
@@ -0,0 +1,86 @@
+use crate::block::{Block, BV};
+use crate::context::ScopeContext;
+use crate::db::{Db, DbKind};
+use crate::everything::Everything;
+use crate::game::GameFlags;
+use crate::item::{Item, ItemLoader};
+use crate::scopes::Scopes;
+use crate::token::Token;
+use crate::tooltipped::Tooltipped;
+use crate::trigger::validate_trigger;
+use crate::validate::validate_modifiers_with_base;
+use crate::validator::{Validator, ValueValidator};
+
+#[derive(Clone, Debug)]
+pub struct LeaseContract {}
+
+inventory::submit! {
+    ItemLoader::Normal(GameFlags::Ck3, Item::LeaseContract, LeaseContract::add)
+}
+
+// TODO: verify somewhere that `theocracy_lease` is a defined item.
+
+impl LeaseContract {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::LeaseContract, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for LeaseContract {
+    fn validate(&self, key: &Token, block: &Block, data: &Everything) {
+        let mut vd = Validator::new(block, data);
+        vd.req_field("hierarchy");
+        vd.field_validated_block("hierarchy", |block, data| {
+            let mut vd = Validator::new(block, data);
+            vd.field_validated_key_block("ruler_valid", |key, block, data| {
+                let mut sc = ScopeContext::new(Scopes::Character, key);
+                validate_trigger(block, data, &mut sc, Tooltipped::No);
+            });
+            vd.field_validated_key_block("liege_or_vassal_valid", |key, block, data| {
+                let mut sc = ScopeContext::new(Scopes::Character, key);
+                sc.define_name("target", Scopes::Character, key);
+                validate_trigger(block, data, &mut sc, Tooltipped::No);
+            });
+            vd.field_validated_key_block("barony_valid", |key, block, data| {
+                let mut sc = ScopeContext::new(Scopes::LandedTitle, key);
+                validate_trigger(block, data, &mut sc, Tooltipped::No);
+            });
+            {
+                let mut sc = ScopeContext::new(Scopes::Character, key);
+                vd.field_target("lessee", &mut sc, Scopes::Character);
+            }
```

Comment:
Not sure why it is enclosed by braces. Perhaps can reuse sc in other field validations too

Reply:
Yeah it's deliberate, to avoid re-using it later.

My habit of re-using sc's in other validators is a bit messy, since they do remember scopes that are set in each use.

However you can't set scopes in a target field so there's no need for the braces here :)